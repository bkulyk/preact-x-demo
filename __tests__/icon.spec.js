import React from 'react';
import Icon from '@bitchin/icon';
import { shallow } from './enzyme';
import { it, expect } from './jest';

it('should render correctly', () => {
  expect(shallow(<Icon icon="favorite" />)).toMatchSnapshot();
});

it('should render correctly with children instead of icon prop', () => {
  expect(shallow(<Icon>favorite</Icon>)).toMatchSnapshot();
});

it('should render correctly with extra props', () => {
  expect(shallow((
    <Icon
      className="custom"
      data-something="something"
    >
      favorite
    </Icon>
  ))).toMatchSnapshot();
});
