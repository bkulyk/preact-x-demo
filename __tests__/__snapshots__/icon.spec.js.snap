// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<i
  className="material-icons "
  title="favorite"
>
  favorite
</i>
`;

exports[`should render correctly with children instead of icon prop 1`] = `
<i
  className="material-icons "
  title="favorite"
>
  favorite
</i>
`;

exports[`should render correctly with extra props 1`] = `
<i
  className="material-icons custom"
  data-something="something"
  title="favorite"
>
  favorite
</i>
`;
